From 42e502d4cd83f7cc7bb5faacdb328af5ec87c397 Mon Sep 17 00:00:00 2001
From: Joel Selvaraj <foss@joelselvaraj.com>
Date: Tue, 27 Aug 2024 21:49:05 -0500
Subject: [PATCH 122/143] ASoC: codecs: cs35l36: add dummy .set_tdm_slot

Required by certain machine drivers like the sound/qcom/sdm845.c. Since the tdm slots are
kind of configured through kcontrol such as "Channel Mux" and "ASPTX1SRC" etc, we simply
introduce a dummy .set_tdm_slot function for now.

Signed-off-by: Joel Selvaraj <foss@joelselvaraj.com>
---
 sound/soc/codecs/cs35l36.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/sound/soc/codecs/cs35l36.c b/sound/soc/codecs/cs35l36.c
index 654aac32d9..c702c0d90e 100644
--- a/sound/soc/codecs/cs35l36.c
+++ b/sound/soc/codecs/cs35l36.c
@@ -950,10 +950,22 @@ static const struct cs35l36_pll_config *cs35l36_get_clk_config(
 	return NULL;
 }
 
+static int cs35l36_set_tdm_slot(struct snd_soc_dai *dai,
+				    unsigned int tx_mask, unsigned int rx_mask,
+				    int slots, int slot_width)
+{
+	// The tdm slots are kind of configured through kcontrol such as
+	// "Channel Mux" and "ASPTX1SRC" etc. Without this function, machine
+	// driver like qcom sdm845 driver will error trying to configure the tdm slots.
+	// Not sure if this is the ideal fix. But this shouldn't cause any harm.
+	return 0;
+}
+
 static const struct snd_soc_dai_ops cs35l36_ops = {
 	.set_fmt = cs35l36_set_dai_fmt,
 	.hw_params = cs35l36_pcm_hw_params,
 	.set_sysclk = cs35l36_dai_set_sysclk,
+	.set_tdm_slot = cs35l36_set_tdm_slot,
 };
 
 #define CS35L36_RATES (		    \
-- 
2.49.0

